plugins/Phaser/PhaserControlDialog.cpp

```diff
@@ -0,0 +1,190 @@
+
+	QPixmap cutoffDotLeftImg = PLUGIN_NAME::getIconPixmap("cutoffDotLeft");
+	m_cutoffDotLeftLabel = new QLabel(this);
+	m_cutoffDotLeftLabel->setPixmap(cutoffDotLeftImg);
+	m_cutoffDotLeftLabel->setAttribute(Qt::WA_TransparentForMouseEvents);
+	m_cutoffDotLeftLabel->move(182, 139);
+
+	QPixmap cutoffDotRightImg = PLUGIN_NAME::getIconPixmap("cutoffDotRight");
+	m_cutoffDotRightLabel = new QLabel(this);
+	m_cutoffDotRightLabel->setPixmap(cutoffDotRightImg);
+	m_cutoffDotRightLabel->setAttribute(Qt::WA_TransparentForMouseEvents);
+	m_cutoffDotRightLabel->move(182, 139);
+
+	LedCheckBox * enableLFO = new LedCheckBox("", this, tr("Enable LFO"), LedCheckBox::Green);
+	enableLFO->setModel(&controls->m_enableLFOModel);
+	enableLFO->move(160, 63);
+	ToolTip::add(enableLFO, tr("Enable LFO"));
+
+	PixmapButton * invertButton = new PixmapButton(this, tr("Invert wet signal"));
+	invertButton->move(276, 6);
+	invertButton->setActiveGraphic(PLUGIN_NAME::getIconPixmap("invert_active"));
+	invertButton->setInactiveGraphic(PLUGIN_NAME::getIconPixmap("invert_inactive"));
+	invertButton->setCheckable(true);
+	invertButton->setModel(&controls->m_invertModel);
+	ToolTip::add(invertButton, tr("Invert wet signal"));
+
+	PixmapButton * wetButton = new PixmapButton(this, tr("Wet"));
+	wetButton->move(242, 6);
+	wetButton->setActiveGraphic(PLUGIN_NAME::getIconPixmap("wet_active"));
+	wetButton->setInactiveGraphic(PLUGIN_NAME::getIconPixmap("wet_inactive"));
+	wetButton->setCheckable(true);
+	wetButton->setModel(&controls->m_wetModel);
+	ToolTip::add(wetButton, tr("Isolate wet signal"));
+
+	EqFader * outFader = new EqFader(&controls->m_outGainModel,tr("Output gain"),
+		this, &controls->m_outPeakL, &controls->m_outPeakR);
+	outFader->setMinimumHeight(84);
+	outFader->setSizePolicy(QSizePolicy::Ignored, QSizePolicy::Ignored);
+	outFader->resize(23, 84);
+	outFader->move(336, 33);
+	outFader->setDisplayConversion(false);
+	outFader->setHintText(tr("Gain"), "dBFS");
+	ToolTip::add(outFader, tr("Output gain"));
+
+	EqFader * inFader = new EqFader(&controls->m_inGainModel,tr("Input gain"),
+		this, &controls->m_inPeakL, &controls->m_inPeakR);
+	inFader->setMinimumHeight(84);
+	inFader->setSizePolicy(QSizePolicy::Ignored, QSizePolicy::Ignored);
+	inFader->resize(23, 84);
+	inFader->move(13, 33);
+	inFader->setDisplayConversion(false);
+	inFader->setHintText(tr("Gain"), "dBFS");
+	ToolTip::add(inFader, tr("Input gain"));
+
+	connect(gui->mainWindow(), SIGNAL(periodicUpdate()), this, SLOT(updateSliders()));
+}
+
+
+void PhaserControlDialog::updateSliders()
+{
+	// Magic. Do not touch.
+	m_cutoffDotLeftLabel->move((int((log2(m_controls->m_effect->getCutoff(0)) - log2(PHA_MIN_FREQ)) /
+		log2(PHA_MAX_FREQ / PHA_MIN_FREQ) * PHA_DOT_SLIDER_LENGTH)) + 12, 139);
+	m_cutoffDotRightLabel->move((int((log2(m_controls->m_effect->getCutoff(1)) - log2(PHA_MIN_FREQ)) /
+		log2(PHA_MAX_FREQ / PHA_MIN_FREQ) * PHA_DOT_SLIDER_LENGTH)) + 12, 139);
```

Comment:
> Also, I'm curious why you chose it instead of log(base e).

The formula used makes the visualizer exponential based on octaves/semitones/cents rather than just using Euler's number.  It'll make the frequency locations placed the same way as you'd see from any spectrum analyzer.

> You may use one log2 for each numerator or two for each denominator.

What do you mean by this?  Are you saying this can be simplified?